src/Features/Core/Portable/PdbSourceDocument/PdbSourceDocumentLoaderService.cs

```diff
@@ -33,8 +35,15 @@ public PdbSourceDocumentLoaderService()
             // Otherwise, check the easiest (but most unlikely) case which is the document exists on the disk
             if (File.Exists(sourceDocument.FilePath))
             {
-                // TODO: Make sure the hash of the file is correct: https://github.com/dotnet/roslyn/issues/57351
-                return Task.FromResult<TextLoader?>(new FileTextLoader(sourceDocument.FilePath, Encoding.UTF8));
+                using var fileStream = new FileStream(sourceDocument.FilePath, FileMode.Open, FileAccess.Read, FileShare.Read | FileShare.Delete);
+                // TODO: Don't hard code UTF8: https://github.com/dotnet/roslyn/issues/57350
+                var sourceText = SourceText.From(fileStream, Encoding.UTF8, sourceDocument.HashAlgorithm);
+                var fileChecksum = sourceText.GetChecksum();
+
+                if (fileChecksum.SequenceEqual(sourceDocument.Checksum))
+                {
+                    return Task.FromResult<TextLoader?>(new FileTextLoader(sourceDocument.FilePath, Encoding.UTF8));
```

Comment:
>FileTextLoader [](http://example.com/codeflow?start=60&length=14)

Since we already read the content of the file above wouldn't it be better to use that content here as well?

Reply:
Yeah, you're right. I was going to leave this for a follow up, as I suspect this API will change when the debugger bits are available, but I should probably just do it now :)